src/main/java/lotto/Calculator.java

```diff
@@ -0,0 +1,32 @@
+package lotto;
+
+import java.math.BigDecimal;
+import java.util.Map;
+
+import lotto.domain.Prize;
+
+public class Calculator {
+    private static final BigDecimal PRICE_PER_LOTTO = BigDecimal.valueOf(1000);
+
+    public double calculateYield(BigDecimal purchaseAmount, BigDecimal prizeMoney) {
+        double earnedMoney = prizeMoney.subtract(purchaseAmount).doubleValue();
+        return 1 + (earnedMoney / purchaseAmount.doubleValue());
+    }
+
+    public int calculatePurchasedLottoNum(BigDecimal payAmount) {
+        return payAmount.divide(PRICE_PER_LOTTO).intValue();
+    }
+
+    public BigDecimal calculatePrizeMoney(Map<Prize, Integer> rankMap) {
+        BigDecimal prizeMoney = BigDecimal.ZERO;
+
+        for (Prize value : Prize.values()) {
+            prizeMoney =
+                    prizeMoney.add(
+                            BigDecimal.valueOf(rankMap.get(value))
+                                      .multiply(value.getPrizeMoney()));
```

Comment:
```suggestion
            BigDecimal bigDecimal = BigDecimal.valueOf(rankMap.get(value));
            prizeMoney = prizeMoney.add(bigDecimal.multiply(value.getPrizeMoney()));
```
코멘트를 모호하게 남겼네요. 변수로 분리하고 가면 조금 더 읽기 쉬운 코드일 것 같습니다.